src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -90,5 +121,100 @@ public override int Execute()
             baseUri + id.ToLowerInvariant() + "/" + version.ToNormalizedString() + "/" + id.ToLowerInvariant() +
             "." +
             version.ToNormalizedString() + ".nupkg";
+
+        public void UpdateWorkloads(bool includePreviews = false)
+        {
+            _reporter.WriteLine();
+            var featureBand = new SdkFeatureBand(string.Join('.', _sdkVersion.Major, _sdkVersion.Minor, _sdkVersion.SdkFeatureBand));
+
+            var workloadIds = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(featureBand);
+            _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(includePreviews).Wait();
+            var manifestsToUpdate = _workloadManifestUpdater.CalculateManifestUpdates();
+
+            UpdateWorkloadsWithInstallRecord(workloadIds, featureBand, manifestsToUpdate);
+
+            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            {
+                _workloadInstaller.GetPackInstaller().GarbageCollectInstalledWorkloadPacks();
+            }
+
+            _reporter.WriteLine();
+            _reporter.WriteLine(string.Format(LocalizableStrings.UpdateSucceeded, string.Join(" ", workloadIds)));
+            _reporter.WriteLine();
+        }
+
+        private void UpdateWorkloadsWithInstallRecord(
+            IEnumerable<WorkloadId> workloadIds,
+            SdkFeatureBand sdkFeatureBand,
+            IEnumerable<(ManifestId manifestId, ManifestVersion existingVersion, ManifestVersion newVersion)> manifestsToUpdate)
+        {
+            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            {
+                var installer = _workloadInstaller.GetPackInstaller();
+                IEnumerable<PackInfo> workloadPackToUpdate = new List<PackInfo>();
+
+                TransactionalAction.Run(
+                    action: () =>
+                    {
+                        foreach (var manifest in manifestsToUpdate)
+                        {
+                            _workloadInstaller.InstallWorkloadManifest(manifest.manifestId, manifest.newVersion, sdkFeatureBand);
+                        }
+
+                        _workloadResolver.RefreshWorkloadManifests();
+
+                        workloadPackToUpdate = workloadIds
+                            .SelectMany(workloadId => _workloadResolver.GetPacksInWorkload(workloadId.ToString()))
+                            .Distinct()
+                            .Select(packId => _workloadResolver.TryGetPackInfo(packId));
+
+                        foreach (var packId in workloadPackToUpdate)
+                        {
+                            installer.InstallWorkloadPack(packId, sdkFeatureBand);
+                        }
+
+                        foreach (var workloadId in workloadIds)
+                        {
+                            _workloadInstaller.GetWorkloadInstallationRecordRepository()
```

Comment:
there is no delete of the old record ? and we don't run a garbage collection after the update?

Reply:
Garbage collection (including deleting the old records) is called here: https://github.com/dotnet/sdk/pull/17642/files#diff-32dff2c2d8df112281b252add6af08593e98425431807590ed2422e93c8bcdb1R138